### PR TITLE
[#348] OpenAPI V3: Preferences page for code templates

### PR DESCRIPTION
**Pull request by ghillairet**
Added Templates page in Preferences.

screen shot 2017-09-21 at 16 55 42

### CODE REVIEW

File: com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/preferences/OpenApiTemplatePreferences.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                getContextTypeRegistry());
+        if (dialog.open() == Window.OK) {
+            return dialog.getTemplate();
+        }
+        return null;
+    }
+
+    protected boolean isShowFormatterSetting() {
+        return false;
+    }
```
Comment (tfesenko) [inline com.reprezen.swagedit.openapi3/src/com/reprezen/swagedit/openapi3/preferences/OpenApiTemplatePreferences.java]: This class is almost identical to `SwaggerTemplatePreferences`. It means that we can extract the reusable code and only leave OAS3-specific config and customizations.

**Comment by tfesenko** [comment]
Passed QA - works exactly as advertised.
Code review: suggested extracting duplicate code into a superclass.

**Comment by ghillairet** [comment]
@tfesenko Done refactoring in https://github.com/RepreZen/KaiZen-OpenAPI-Editor/pull/389/commits/e6f36a28167e131b9716259fc1501f0b2957c941

**Comment by tfesenko** [comment]
Thanks! Looks good!

**Pull request merged**